core/controllers/profile_test.py

```diff
@@ -420,342 +545,286 @@ def test_email_preferences_updates(self):
-            response['profile_picture_data_url'], 'new_profile_picture')
-        self.logout()
+class RemoveAccountPageTests(test_utils.GenericTestBase):
 
+    def setUp(self):
+        super(RemoveAccountPageTests, self).setUp()
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.login(self.EDITOR_EMAIL)
 
-class UserContributionsTests(test_utils.GenericTestBase):
+    def test_get_remove_account_page(self):
+        with self.swap(constants, 'ENABLE_ACCOUNT_REMOVAL', True):
+            response = self.get_html_response('/remove-account')
+            self.assertIn(
+                '<remove-account-page></remove-account-page>', response.body)
 
-    USERNAME_A = 'a'
-    EMAIL_A = 'a@example.com'
-    USERNAME_B = 'b'
-    EMAIL_B = 'b@example.com'
-    EXP_ID_1 = 'exp_id_1'
+    def test_get_remove_account_page_disabled(self):
+        with self.swap(constants, 'ENABLE_ACCOUNT_REMOVAL', False):
+            self.get_html_response('/remove-account', expected_status_int=404)
 
-    def test_null_case(self):
-        # Check that the profile page for a user with no contributions shows
-        # that they have 0 created/edited explorations.
-        self.signup(self.EMAIL_A, self.USERNAME_A)
-        response_dict = self.get_json(
-            '/profilehandler/data/%s' % self.USERNAME_A)
-        self.assertEqual(
-            response_dict['created_exp_summary_dicts'], [])
-        self.assertEqual(
-            response_dict['edited_exp_summary_dicts'], [])
 
-    def test_created(self):
-        # Check that the profile page for a user who has created
-        # a single exploration shows 1 created and 1 edited exploration.
-        self.signup(self.EMAIL_A, self.USERNAME_A)
-        user_a_id = self.get_user_id_from_email(self.EMAIL_A)
-        user_a = user_services.UserActionsInfo(user_a_id)
-        self.save_new_valid_exploration(
-            self.EXP_ID_1, user_a_id, end_state_name='End')
-        rights_manager.publish_exploration(user_a, self.EXP_ID_1)
+class RemoveAccountHandlerTests(test_utils.GenericTestBase):
 
-        response_dict = self.get_json(
-            '/profilehandler/data/%s' % self.USERNAME_A)
+    def setUp(self):
+        super(RemoveAccountHandlerTests, self).setUp()
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.login(self.EDITOR_EMAIL)
 
-        self.assertEqual(len(
-            response_dict['created_exp_summary_dicts']), 1)
-        self.assertEqual(len(
-            response_dict['edited_exp_summary_dicts']), 1)
-        self.assertEqual(
-            response_dict['created_exp_summary_dicts'][0]['id'],
-            self.EXP_ID_1)
-        self.assertEqual(
-            response_dict['edited_exp_summary_dicts'][0]['id'],
-            self.EXP_ID_1)
+    def test_post_remove_account(self):
+        csrf_token = self.get_new_csrf_token()
+        self.assertEqual(self.post_json(
+            '/removeaccounthandler', {}, csrf_token=csrf_token), {})
```

Comment:
Er -- presumably this should test a bit more than that. Some side effect should happen? Like, data should be deleted?

Reply:
I want to implement that in the next PR (first deletion steps), I don't want to make the PRs that big, I could also remove the handler from this PR but we would still have a page that doesn't do anything (but is hidden behind a flag).